test/g_searcher/search_results/search_result_parser_test.exs

```diff
@@ -0,0 +1,85 @@
+defmodule GSearcher.SearchResults.SearchResultParserTest do
+  use GSearcher.DataCase, async: true
+
+  alias GSearcher.SearchResults.SearchResultParser
+
+  describe "parse/1" do
+    test "extracts advertiser information given valid HTML" do
```

Comment:
This function also returns a result and HTML cache how about update text description a bit
```suggestion
    test "extracts search result when given valid HTML" do
```